opal/runtime/opal_progress.c

```diff
@@ -474,7 +474,7 @@ static int _opal_progress_unregister (opal_progress_callback_t cb, volatile opal
         (void) opal_atomic_swap_ptr ((opal_atomic_intptr_t *) (callback_array + i), (intptr_t) callback_array[i+1]);
     }
 
-    callback_array[*callback_array_len] = fake_cb;
+    callback_array[*callback_array_len - 1] = fake_cb;
```

Comment:
I would just swap the 2 lines, to avoid replicating the decrement operation.

Reply:
That was my first solution but I went for this in the interest of readability. Changed it back to just swapping the two lines.